internal/controllers/users.go

```diff
@@ -0,0 +1,171 @@
+package controllers
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/cyverse/QMS/internal/model"
+	"github.com/labstack/echo"
+)
+
+// swagger:route GET admin/users adm in listUsers
+// Returns a List all the Users By Admin
+// responses:ssw
+//   200: UserResponse
+//   404: RootResponse
+func (s Server) GetAllUsers(ctx echo.Context) error {
+	data := []model.Users{}
+	err := s.GORMDB.Debug().Find(&data).Error
+	if err != nil {
+		return ctx.JSON(http.StatusBadRequest, map[string]interface{}{"error": err.Error()})
+	}
+	return ctx.JSON(http.StatusOK, model.SuccessResponse(data, http.StatusOK))
+}
+
+type PlanDetails struct {
+	UserId string
+	Name   string
+	Usage  string
+	Quota  float64
+	Unit   string
+}
+type QuotaDetails struct {
+	PlanName     string
+	Quota        float64
+	ResourceName string
+	Unit         string
+}
+
+type UsageDetails struct {
+	PlanName     string
+	ResourceName string
+	Usage        string
+	Unit         string
+}
+
+// swagger:route GET /users/{UserName}/plan users listUserPlansByID
+// Returns a List all the User Plan Details
+// responses:
+//   200: UserResponse
+//   404: RootResponse
+func (s Server) GetUserPlanDetails(ctx echo.Context) error {
+	username := ctx.Param("username")
+	if username == "" {
+		return ctx.JSON(http.StatusBadRequest, model.ErrorResponse("Invalid Usernaem", http.StatusBadRequest))
```

Comment:
Looks like there's a typo here as well:

```suggestion
		return ctx.JSON(http.StatusBadRequest, model.ErrorResponse("Invalid Username", http.StatusBadRequest))
```